modules/terraform/options.go

```diff
@@ -102,3 +106,144 @@ func WithDefaultRetryableErrors(t *testing.T, originalOptions *Options) *Options
+// Will return an error if GetAllVariablesFromVarFileE returns an error or the key provided does not exist in the file.
+// For list or map, use GetVariableAsListFromVarFile or GetVariableAsMapFromVarFile, respectively.
+func GetVariableAsStringFromVarFileE(t *testing.T, option *Options, fileName string, key string) (string, error) {
+	variables, err := GetAllVariablesFromVarFileE(t, option, fileName)
+
+	if err != nil {
+		return "", err
+	}
+
+	variable, exists := variables[key]
+
+	if !exists {
+		return "", InputFileKeyNotFound(key)
+	}
+
+	return fmt.Sprintf("%v", variable), nil
+}
+
+// GetVariableAsMapFromVarFile Gets the map represention of a variable from a provided input file found in VarFile
+func GetVariableAsMapFromVarFile(t *testing.T, option *Options, fileName string, key string) map[string]string {
+	result, err := GetVariableAsMapFromVarFileE(t, option, fileName, key)
+	require.NoError(t, err)
+
+	return result
+}
+
+// GetVariableAsMapFromVarFileE Gets the map represention of a variable from a provided input file found in VarFile
+// Returns an error if GetAllVariablesFromVarFileE returns an error, the key provided does not exist, or the value associated with the key is not a map
+func GetVariableAsMapFromVarFileE(t *testing.T, option *Options, fileName string, key string) (map[string]string, error) {
+	resultMap := make(map[string]string)
+	variables, err := GetAllVariablesFromVarFileE(t, option, fileName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	_, exists := variables[key]
+
+	if !exists {
+		return nil, InputFileKeyNotFound(key)
+	}
+
+	if reflect.TypeOf(variables[key]).String() != "[]map[string]interface {}" {
+		return nil, UnexpectedOutputType{Key: key, ExpectedType: "map", ActualType: reflect.TypeOf(variables[key]).String()}
+	}
+
+	mapKeys := variables[key].([]map[string]interface{})[0]
+
+	for mapKey, mapVal := range mapKeys {
+		resultMap[mapKey] = fmt.Sprintf("%v", mapVal)
+	}
+
+	return resultMap, nil
+}
+
+// GetVariableAsListFromVarFile Gets the string list represention of a variable from a provided input file found in VarFile
+func GetVariableAsListFromVarFile(t *testing.T, option *Options, fileName string, key string) []string {
+	result, err := GetVariableAsListFromVarFileE(t, option, fileName, key)
+	require.NoError(t, err)
+
+	return result
+}
+
+// GetVariableAsListFromVarFileE Gets the string list represention of a variable from a provided input file found in VarFile
+// Will return error if GetAllVariablesFromVarFileE returns an error, the key provided does not exist, or the value associated with the key is not a list
+func GetVariableAsListFromVarFileE(t *testing.T, option *Options, fileName string, key string) ([]string, error) {
+	resultArray := []string{}
+	variables, err := GetAllVariablesFromVarFileE(t, option, fileName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if _, exists := variables[key]; !exists {
```

Comment:
Same NIT here about storing the return value instead of looking it up again several times below.